### PR TITLE
normalize base of language filter pattern to support clients on another OS

### LINKED ISSUE #8261: Relative Patterns don't work accross Platforms
### Bug Description:
When running the back end on linux and running the front end on Windows, document selectors using relative paths never match: For example, the CodeLensProvider for the Java Test Runner plugin never gets invoked when opening a Java file with a test in it. This works when both back-end and front-end run on the same platform

### Steps to Reproduce:

1. Install the Java Test Runner plugin (https://marketplace.visualstudio.com/items?itemName=vscjava.vscode-java-test)
2. Run Theia back-end on linux, open the browser on Windows
3. Clone https://github.com/che-samples/console-java-simple
4. Open the Singe file containing tests.
4. Observe: after initializing all plugins (you'll need java, java debug), you do not get a code lens saysing "Debug Test | Run Test".

### Additional Information

The problem arises that monaco compares the currently opened file with a relative glob pattern in the DocumentSelector by using URI.fsPath, which in turn calls this method: https://github.com/microsoft/vscode/blob/90efeb4f2dc41827b382c8bb5da1c3abfff9459f/src/vs/base/common/uri.ts#L579
This results in the path separators being converted to backslashes. However, the same treatment is not done with the pattern, so the paths will not match. For example, the open file might be file:///projects/foo/bar and the pattern we match might be /projects/foo/bar, but we'll end up comparing \projects\foo\bar with /projects/foo/bar, which obviously fails.
On VS Code this does not arise, since they're always running on the same machine.
The problem seems fixed in the master version of vs code: https://github.com/microsoft/vscode/blob/c86411397d528d96686345d94f0478fbd77cc0b2/src/vs/editor/common/modes/languageSelector.ts#L96

### PR DESCRIPTION
**Pull request by akosyakov**
#### What it does

- fix #8261: normalize pattern base to support  clients on another OS
- it also removes pulling the plugin host code into the browser process

#### How to test

- deploy Theia on Linux and test on Windows client (on Linux as well for regressions)
  - gitpod backend is Linux if someone with Windows machine wants to try quickly
- install Java Test Runner extension from `Extensions` view
- check out and open https://github.com/che-samples/console-java-simple
- run code lenses

#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

**Comment by akosyakov** [comment]
@eclipse-theia/ecd-theia-committers Can someone with Windows machine help to verify? Backend should run on Linux (you just use Gitpod).

File: packages/plugin-ext/src/main/browser/languages-main.ts [hunk 1]
```diff
@@ -40,7 +40,6 @@ import {
     MarkerSeverity, DocumentLink, WorkspaceSymbolParams, CodeAction, CompletionDto
 } from '../../common/plugin-api-rpc-model';
 import { RPCProtocol } from '../../common/rpc-protocol';
-import { fromLanguageSelector } from '../../plugin/type-converters';
```
[the remark this answers is not in the document]
Comment (akosyakov) [reply packages/plugin-ext/src/main/browser/languages-main.ts:43]: It seems to be still used inside the plugin host process: https://github.com/eclipse-theia/theia/blob/27720111a1f6dc3375ec5f0d099db31aa9141618/packages/plugin-ext/src/plugin/plugin-context.ts#L560

[the remark this answers is not in the document]
Comment (akosyakov) [reply packages/plugin-ext/src/main/browser/languages-main.ts:43]: removed

**Comment by lmcbout** [comment]
Code wise looks good, Anyone with a window environment can perform a last check before merging for the release would be appreciated. Thanks

**Comment by akosyakov** [comment]
Thomas already checked on windows against linux: https://github.com/eclipse-theia/theia/pull/8268#pullrequestreview-458224079

**Comment by lmcbout** [comment]
Good, You should merge it
Thanks @akosyakov

**Pull request merged**